src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -296,11 +299,62 @@ public override async Task<ProvideSemanticTokensResponse> ProvideSemanticTokensA
                 semanticTokensParams,
                 cancellationToken).ConfigureAwait(false);
 
-            var result = new ProvideSemanticTokensResponse(csharpResults, csharpDoc.HostDocumentSyncVersion);
+            var result = new ProvideSemanticTokensResponse(csharpResults, editResult: null, csharpDoc.HostDocumentSyncVersion);
 
             return result;
         }
 
+        [Obsolete]
+#pragma warning disable CS0809 // Obsolete member overrides non-obsolete member
+        public override async Task<ProvideSemanticTokensResponse> ProvideSemanticTokensEditsAsync(
+#pragma warning restore CS0809 // Obsolete member overrides non-obsolete member
+            SemanticTokensEditsParams semanticTokensEditsParams,
+            CancellationToken cancellationToken)
+        {
+            if (semanticTokensEditsParams is null)
+            {
+                throw new ArgumentNullException(nameof(semanticTokensEditsParams));
+            }
+
+            if (!_documentManager.TryGetDocument(semanticTokensEditsParams.TextDocument.Uri, out var documentSnapshot))
+            {
+                return null;
+            }
+
+            if (!documentSnapshot.TryGetVirtualDocument<CSharpVirtualDocumentSnapshot>(out var csharpDoc))
+            {
+                return null;
+            }
+
+            semanticTokensEditsParams.TextDocument.Uri = csharpDoc.Uri;
+
+            var csharpResults = await _requestInvoker.ReinvokeRequestOnServerAsync<SemanticTokensEditsParams, SumType<LanguageServer.Protocol.SemanticTokens, SemanticTokensEdits>>(
```

Comment:
This seems potentially wrong, although it works in VS. I initially tried returning the O# equivalent of SemanticTokens and SemanticTokensEdits via `SumType<SemanticTokens, SemanticTokensDelta>`. However, this always returned `SemanticTokens` even if C# returned to us a type of `SemanticTokensEdits`. If this needs to be modified, let me know!

Reply:
Hmm, are SemanticTokens and SemanticTokensEdits compatible from a serialization perspective?

Reply:
@allisonchou what part about it feels wrong to you?

Reply:
> I don't think so, they both have pretty different internal contents/properties.

Hmm maybe I'm missing something, why have the SumType then? Will this ever return `SemanticTokens`?

Reply:
Whoops missed this! Admittedly I'm not too familiar with serialization, I initially assumed that they wouldn't be compatible due to the difference in properties but now I feel like that assumption was wrong. Roslyn returns `SumType<SemanticTokens, SemanticTokensEdits>` back to LSP so I'm guessing they would have to be compatible for that to work.